package.json

```diff
@@ -1,14 +1,19 @@
 {
   "license": "Apache-2.0",
   "private": true,
-  "workspaces": [
-    "packages/client",
-    "packages/common",
-    "packages/main"
-  ],
+  "workspaces": {
+    "packages": [
+      "packages/client",
+      "packages/common",
+      "packages/main"
+    ],
+    "nohoist": [
+      "**/*"
+    ]
+  },
   "scripts": {
     "diagram": "unflatten -l 3 assets/architecture.dot | dot -Tsvg -o assets/architecture.svg",
-    "postinstall": "scripts/postinstall",
+    "prepare": "scripts/prepare",
```

Comment:
I changed some scripts from `postinstall` to `prepare` because `prepare` runs after installing dependencies, but does not run when the package itself is installed as a dependency of another package. `postinstall` runs in both cases.